### PR TITLE
[example] Upgrade pbf2d.py to new syntax

### PR DESCRIPTION
**Pull request by archibate**
Related issue = WeChat (private)

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: examples/pbf2d.py [hunk 1]
```diff
@@ -232,90 +215,60 @@ def compute_position_deltas():
...
-    find_particle_neighbors()
+    prologue()
     for _ in range(pbf_num_iters):
-        compute_lambdas()
-        compute_position_deltas()
-        apply_position_deltas()
-
-    confine_to_boundary()
-    update_velocities()
-    # no vorticity/xsph because we cannot do cross product in 2D...
```
Comment (archibate) [inline examples/pbf2d.py:274]: We can do cross 2D now, so what should I do?

File: examples/pbf2d.py [hunk 2]
```diff
@@ -46,21 +46,21 @@ def round_up(f, s):
...
+poly6_factor = 315.0 / 64.0 / math.pi
+spiky_grad_factor = -45.0 / math.pi
 
-old_positions = ti.Vector(dim, dt=ti.f32)
-positions = ti.Vector(dim, dt=ti.f32)
-velocities = ti.Vector(dim, dt=ti.f32)
+old_positions = ti.Vector.field(dim, float)
+positions = ti.Vector.field(dim, float)
+velocities = ti.Vector.field(dim, float)
 # Once taichi supports clear(), we can get rid of grid_num_particles
```
Comment (archibate) [inline examples/pbf2d.py:55]: What do you mean by `clear()`?

Comment (Rullec) [reply examples/pbf2d.py:55]: Maybe we need to ask the author of this example...

File: examples/pbf2d.py (hunk 1 above)
Comment (k-ye) [reply examples/pbf2d.py:274]: Nothing really. We need the direction, not the magnitude.

File: examples/pbf2d.py [hunk 3]
```diff
@@ -194,7 +181,7 @@ def find_particle_neighbors():
 
 
 @ti.kernel
-def compute_lambdas():
+def substep():
```
Comment (k-ye) [inline examples/pbf2d.py:184]: Actually, would you not merge them into a single kernel? I wrote the names to make the computation stages obvious.

File: examples/pbf2d.py (hunk 2 above)
Comment (k-ye) [reply examples/pbf2d.py:55]: I wanted to use `dynamic` initially, but `dense` seems to work OK. So I think we can remove this comment now.

```suggestion
```

File: examples/pbf2d.py (hunk 3 above)
Comment (archibate) [reply examples/pbf2d.py:184]: Why not add some comments per-offload instead? I just want to export them to [Taichi.js](https://taichi-dev.github.io/taichi.js) easier..

**Review by k-ye — APPROVED** [review]
Thank you for the help!

File: examples/pbf2d.py (hunk 3 above)
Comment (k-ye) [reply examples/pbf2d.py:188]: I see. Sorry i don't know much about js, is there a limitation for how many kernels we can have there?

Comment (archibate) [reply examples/pbf2d.py:188]: Just hard to seek out the launch order..

**Pull request merged**